### PR TITLE
Adding initial recipe for castorapi (python)

### PR DESCRIPTION
**Pull request by wouterpotters**
@conda-forge/help-python this PR is ready for review.
@conda-forge-admin, please ping team 
@conda-forge-admin, please ping @conda-forge/help-python
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by wouterpotters** [comment]
Ready to merge as far as I can tell.

File: recipes/castorapi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 4544ab11e1144c1a66a466f5ffc379ff1031babe6d74c0fba027f3d4fd6d3b0c
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
```
Comment (npavlovikj) [inline recipes/castorapi/meta.yaml:15]: I believe this recipe can be `noarch`, so please add `noarch: python`, https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages.

File: recipes/castorapi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+requirements:
+  host:
```
Comment (npavlovikj) [inline recipes/castorapi/meta.yaml:17]: In general, `pip` and `python` are the only dependencies that need to be in the `host` sections, while the remaining ones go to `run`.

File: recipes/castorapi/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  imports:
+    - castorapi
+
+about:
+  home: "https://github.com/wouterpotters/castorapi"
+  license: MIT
+  license_family: MIT
+  license_file: "LICENSE"
+  summary: "Python API wrapper for Castor EDC to fetch data from you clinical study."
+  doc_url: "https://github.com/wouterpotters/castorapi/README.md"
```
Comment (npavlovikj) [inline recipes/castorapi/meta.yaml:41]: The `doc_url` is not found. You can probably use `https://github.com/wouterpotters/castorapi#usage` instead.

File: recipes/castorapi/meta.yaml (hunk 2 above)
Comment (npavlovikj) [reply recipes/castorapi/meta.yaml:17]: I just saw that you are using:
```
  install_requires=[            # I get to this in a second
            'io',
            'json',
            'os',
            'pandas',
            'requests',
            'progressbar',
            'logging'
      ],
```
in `setup.py`. In that case, you can leave the remaining dependencies here.

Comment (wouterpotters) [reply recipes/castorapi/meta.yaml:18]: (changed nothing in the requirements now; thought that's what you meant)

File: recipes/castorapi/meta.yaml (hunk 1 above)
Comment (wouterpotters) [reply recipes/castorapi/meta.yaml:16]: fixed

File: recipes/castorapi/meta.yaml (hunk 3 above)
Comment (wouterpotters) [reply recipes/castorapi/meta.yaml]: fixed

**Comment by npavlovikj** [comment]
Thank you for the fixes and the contribution @wouterpotters 🎉

**Pull request merged**